torch/_dynamo/variables/distributed.py

```diff
@@ -107,6 +107,9 @@ def is_placement_type(value):
 
```

Comment:
@wanchaol Can you remind me why we needed this variable?

Reply:
This boils down to the "custom fx type" that we've been discussing. So fundamentally we need to inline `Placement` and `DeviceMesh` as a constant in the closure function (i.e. `from_local`) and put that closure function to the fx graph as a call_function node. 

But if we don't have this `PlacementClassVariable`,  dynamo would trace the DTensor's metadata construction as a UserDefinedClass/Object (i.e. `Shard(1)`). UDTs are not constant variable (as it's hard to tell whether a UDT is a ConstantVariable unless user explicitly tell dynamo the objects are ConstantVariable). So this `PlacementClassVariable` allow us to turn the `Shard(1)` as a `PlacementVariable` (which is a constant variable), and then the sharding metadata can be inlined as a closure